docs/coding-guidelines/javascript.md

```diff
@@ -342,22 +371,47 @@ if ( object && 'desired' in object ) { ... }
 if ( object && object.desired ) { ... }
 ```
 
+Note that the `in` operator checks all inherited properties of an object prototype, which can lead to some unexpected scenarios:
+
+```js
+'valueOf' in {}; // true
+```
+
+If testing the presence of a object key using variable input, it's recommended that you use [`Object#hasOwnProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) instead.
+
+```js
+const key = 'valueOf';
+{}.hasOwnProperty( key ); // false
+```
+
 ## Strings
 
 Use single-quotes for string literals:
 
 ```js
-var myStr = 'strings should be contained in single quotes';
+const myStr = 'strings should be contained in single quotes';
 ```
 
 When a string contains single quotes, they need to be escaped with a backslash (\\):
 
 Double quotes can be used in cases where there is a single quote in the string or in JSX attributes.
```

Comment:
Consistency seems subjective here. _What_ are we being consistent about: quote types or escaping quotes?

If we're consistent about quote types then we're basically requiring that we never use double-quotes, except for with JSX.

If we're consistent about escaping, we're saying never escape quotes (except maybe the back-tick inside of template literals).